src/palace/manager/celery/tasks/playtime_entries.py

```diff
@@ -238,22 +237,24 @@ def generate_playtime_report(
                 )
 
 
-def _fetch_distinct_data_source_names_in_range(
-    session: Session, start: date, until: date
-) -> Query[str]:
-    return session.query(
-        select(
-            distinct(PlaytimeSummary.data_source_name),
+def _fetch_distinct_eligible_data_source_names(
+    session: Session,
+) -> list[str]:
+    query = (
+        select(Collection)
+        .join(
+            IntegrationConfiguration,
+            Collection.integration_configuration_id == IntegrationConfiguration.id,
         )
         .where(
-            and_(
-                PlaytimeSummary.timestamp >= start,
-                PlaytimeSummary.timestamp < until,
+            IntegrationConfiguration.protocol.in_(
+                [OPDS2WithODLApi.label(), OPDSForDistributorsAPI.label()]
             )
         )
-        .order_by(PlaytimeSummary.data_source_name)
-        .subquery()
     )
+    ds_names = [c[0].data_source.name for c in session.execute(query).all()]
```

Comment:
Minor: Getting the datasource names this way probably results in n+1 db roundtrips. It would probably be more efficient to just get the datasource names using `joinedload` in a single db query (starting from the one we already have above).

Reply:
The problem is that Collection.data_source is a calculated property of collection.  There is no data_source_id associated with the collection.  But I can use the `joinedload` on IntegrationConfiguration to ensure that it is in memory for the property calculation which I believe will solve the n+1 issue.